src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs

```diff
@@ -2863,13 +2863,29 @@ private object GetDataFromHGLOBAL(string format, IntPtr hglobal)
                     {
                         data = ReadBitmapSourceFromHandle(hglobal);
                     }
-                    //  Restrict deserialization to only primitives 
-                    // and strings here to prevent potentially malicious objects from
+                    // Limit deserialization to only primitive types, which consist of the following:
+                    //
+                    // DataFormats.CommaSeparatedValue
+                    // DataFormats.FileDrop
+                    // DataFormats.Html
+                    // DataFormats.OemText
+                    // DataFormats.PenData
+                    // DataFormats.Rtf
+                    // DataFormats.Serializable
+                    // DataFormats.Text
+                    // DataFormats.UnicodeText
+                    // DataFormats.WaveAudio
+                    // DataFormats.Xaml
+                    // DataFormats.XamlPackage 
+                    // DataFormats.StringFormat
+                    // 
+                    // Out of these primitive types, we will disallow deserialization of 
+                    // DataFormats.StringFormat to prevent potentially malicious objects from
                     // being deserialized as part of a "text" copy-paste or drag-drop.
```

Comment:
Do you actually *disallow* StringFormat or are you just restricting it to safe types? The PR doesn't seem to contain any "disallow" logic and the existing code just seems to use `formatter.Binder = new TypeRestrictingSerializationBinder();` so I don't quite see how that disallows `DataFormats.StringFormat`.

Might want to fix comment to be more clear what actually happens.

Reply:
I agree that the comment is not accurate. Here's some context on the scenario:

DataFormats is just the "bucket" in which the Clipboard data is set. Nothing enforces the type of data that is put in each bucket, so an application can choose to put a serialized object (of any type, not just a string) in the StringFormat bucket. That doesn't mean WPF should deserialize such objects, a malicious object could be serialized and put under the StringFormat object, for example, and WPF would then deserialize it. The way TypeRestrictingSerializationBinder works is, it throws when attempting to deserialize anything other than strings and primitives. This functionality was behind a compat flag, which is being removed in this PR.


The comment is mixing up the concepts of DataFormats and types.

Reply:
Thanks @miguep and @weltkante. I'll update the comment.